Replace Gulp & Bower with npm

coreymcollins opened the pull request:
### DESCRIPTION ###
- [x] Completely removes Gulp and Bower in favor of npm
- [x] Updates jQuery to vanilla JavaScript*
    - [x] Excludes Slick Carousel which seemed like a bear to do, and we might be replacing it as a carousel anyway... so I didn't too in the weeds with it.
- [x] Updates the structure of the theme with `src` and `dist` directories
- [x] Resolves SassLint issues

There are a handful of SassLint issues still needing to be resolved, but I wanted to get the ball rolling on a PR and update progress along the way.

I'll post the full PR update once things are 💯.

coreymcollins committed:
Updates wd_s to use npm, removing Gulp and bower

coreymcollins committed:
Adds Slick

coreymcollins committed:
Updates script paths

coreymcollins committed:
Moves JS files into a /concat/ directory

coreymcollins committed:
Removes the scaffolding JS and uses the accordion JS we already have in place for show/hide

coreymcollins committed:
Updates uglify path for /concat/ folder

coreymcollins committed:
Adds the babel-polyfill file and uglifies it

coreymcollins committed:
Updates imagemin to find all top-level images

coreymcollins committed:
Updates gutenberg file paths

coreymcollins committed:
Adds gutenberg block style scripts

coreymcollins committed:
Updates image paths

coreymcollins committed:
Updates main theme stylesheet

coreymcollins committed:
Fixes relative paths for SVG backgrounds

coreymcollins committed:
Fixes an issue with multiple clicks firing on the backend

coreymcollins committed:
Removes an empty space

coreymcollins committed:
Removes an unncessary file

coreymcollins committed:
ESLint fixes

coreymcollins committed:
Fixes an issue with ACF display options array breaking if no font color is set

coreymcollins committed:
Updates script enqueue paths

coreymcollins committed:
Adds customizer action to package file

coreymcollins committed:
Moves the customizer JS file

coreymcollins committed:
Fixes the order on some table CSS for responsiveness

coreymcollins committed:
Adds a fix to table.js in the event that a table has TH but no THEAD

coreymcollins committed:
Converts table.js to plain JS

coreymcollins committed:
Converts accordion jQuery to vanilla JS

coreymcollins committed:
Converts header button jQuery to vanilla JS

coreymcollins committed:
Updates window ready/body class from jQuery to vanilla JS

coreymcollins committed:
Converts the mobile menu jQuery to vanilla JS

coreymcollins committed:
Checks for table headings before trying to use them

coreymcollins committed:
Bails if the header search form isn’t being used

coreymcollins committed:
Replaces video playback jQuery with vanilla JS

coreymcollins committed:
Converts primary nav jQuery to vanilla JS

coreymcollins committed:
Converts off-canvas jQuery to vanilla JS

coreymcollins committed:
Converts modal jQuery to JS

coreymcollins committed:
ESLint cleanup

coreymcollins committed:
Adds some conditional checks to the header site action

coreymcollins committed:
Replaces customizer jQuery with vanilla JS

coreymcollins committed:
Updates some sass lint items

coreymcollins committed:
Adds an .eslintignore file to ignore the babel polyfill file

coreymcollins committed:
Updates linting location and rules

coreymcollins committed:
Checks for off canvas screen before trying to do anything else

gregrickaby commented:
![screenshot](https://media.giphy.com/media/RiuNdiJcPi8RswW0ay/giphy.gif)

coreymcollins committed:
Updates JS linting config

coreymcollins committed:
Adds stylelint-order plugin

coreymcollins committed:
Adds stylelint rules

coreymcollins committed:
Update stylelint to look for our config file (for now, at least)

coreymcollins committed:
Updates some stylelint rules

coreymcollins committed:
stylelint fixes

coreymcollins committed:
Cleans up some ESLint issues

coreymcollins commented:
This should be ready for testing!

Some simple instructions for testing –
1. CD into the theme
2. Run `npm install` to install all the things
3. Run `npm run watch` to watch all the things, or `npm run build:X` where X can be `css` or `js`
4. Report back any issues you have getting things up and running, compiling, watching, sniffing and linting, etc.

You can see the full breadth of what `npm` actions are available in `package.json` if you want to see more of what's going on.

For testing, please test:
1. Compiling Sass
2. Compiling JS
3. Adding image and icons for image optimization
4. Generating language files
5. Try to break obvious things in PHP, Sass, and JS to generate linting errors/warnings

gregrickaby committed:
dependency bump

gregrickaby committed:
fixes the svg sprite and prettier error

gregrickaby committed:
bump to v12

gregrickaby committed:
readme bump to reflect npm scripts

gregrickaby commented:
My initial pass gets a 👍🏻... 

I did push up a few quick fixes from issues I experience on my local, and bumped the README.

gregrickaby committed:
bump to latest wds coding standards

gregrickaby committed:
watch php files too

gregrickaby committed:
add container to footer

gregrickaby committed:
introduce left-tabbed block options layout

gregrickaby commented:
Tried to break this PR again today by styling the footer and... everything seems to be fine. 

I also reconfigured our blocks to use the left-tabbed interface, which greatly improves the UX while inside the new block editor:
![screenshot](https://dl.dropbox.com/s/m1acptdqffoengt/kapture%202020-02-17%20at%2011.36.10.gif?dl=0)

The only thing left to do is: **make sure WDS Coding Standards are being installed correctly and the linting in `package.json` is pointing at the new rulesets.** cc @aubreypwd 

Once that's done, I can setup the Status Checks in Buddy.

aubreypwd commented:
I will add this to my list of things to do on Friday

vegasgeek commented:
I added ticket #508 
I'm using this branch locally.

I see @gregrickaby's gif above and it shows the block editor being nice and wide. Mine is wedged on the right side of the screen. I provided a link to a screenshot of what I'm seeing. Happy to screenshare if it would be useful.

gregrickaby commented:
@vegasgeek Looks like this is a CSS issue (or more like oversight) with ACF.

If I add:

```css
.acf-tab-wrap.-left .acf-tab-group li a {
    word-wrap: break-word;
}
```
it's not _as_ terrible (but still pretty bad):
![screenshot](https://dl.dropbox.com/s/pbrbmmarn9gcd1u/Screenshot%202020-02-21%2011.01.45.png?dl=0)

I wish I could increase the width of the sidebar in the post editor... 😞

vegasgeek commented:
@gregrickaby in your gif from 4 days ago, it's showing a nice wide version of the block with plenty of room for the side tabs. Is there a trick to moving to that layout rather than having the block settings smashed to the right side column?

gregrickaby commented:
@vegasgeek There is:

![screenshot](https://dl.dropbox.com/s/cv436e7qtscbhe2/Screenshot%202020-02-21%2011.58.04.png?dl=0)

vegasgeek commented:
WHO KEEPS HIDING THESE THINGS RIGHT IN FRONT OF ME?

Just an FYI, still getting a little bit of CSS funkiness, but that widescreen edit definitely makes this usable. Thanks for that.

![Screen Shot 2020-02-21 at 10 02 28 AM](https://user-images.githubusercontent.com/367562/75059214-55851e80-5491-11ea-9a67-22d3906c0e34.png)

coreymcollins commented:
@superfein – the checkbox issue you're reporting looks to be happening on the `master` branch as well, so doesn't seem specific to this pull request. Could you add an issue with your details there rather than tracking it here?

Thanks!

aubreypwd commented:
I was out last Friday, still on my list for this Friday

coreymcollins commented:
Going to take a look at the prettier issues noted above. Greg added prettier with one of his last updates, so I think those are fairly new and weren't caught before.

@aubreypwd – since we're pointing to the WDS Coding Standards for ESLint configs, I'm wondering if we can tie this in there as well? The hope is to move away from having any config files in the theme itself and instead have them in our package(s) for ease of updating and consistency.

aubreypwd commented:
> @aubreypwd – since we're pointing to the WDS Coding Standards for ESLint configs, I'm wondering if we can tie this in there as well? The hope is to move away from having any config files in the theme itself and instead have them in our package(s) for ease of updating and consistency.

You have an:

```json
  "eslintConfig": {
    "extends": [
      "@webdevstudios/js-coding-standards"
    ]
  },
```

in your `package.json` file, which is perfect. I have little experience with prettier so may need your input on that. 

> Getting errors when running npm run build:js in the terminal.

https://github.com/WebDevStudios/wd_s/pull/507#issuecomment-590113956

These issues are coming from the linters for JS because, actually, these files do not meet the modern standards. We may need to go through these files (once I get proper JS linting installed) and correct these @coreymcollins 

**Correction, these ARE coming from prettier, and they are giving suggestions that aren't a part of our coding standards from what I can tell.**

coreymcollins commented:
I'm not super duper familiar with Prettier myself, @gregrickaby added that when he was running through his testing and it seems like it throws a lot of errors/warnings (as seen in the #507 comment) that conflict with rules we have set (i.e., Prettier wants `( '.class-name' )` to become `('.class-name')` with no spaces).

It seems like Prettier offers far less customization than ESLint and I'm wondering if we truly need both?

aubreypwd commented:
> It seems like Prettier offers far less customization than ESLint and I'm wondering if we truly need both?

The last time I was working on Prettier was with Jo when he was here and we found that we were having a hard time getting it to team up with WDS-Coding-Standards (the old one) too, but I do like the fix-it nature of it and still want to bring it in if we can!

aubreypwd commented on package.json:
```diff
@@ -2,53 +2,108 @@
+    "watch:images": "onchange \"src/images\" -- run-s build:images",
+    "watch:php": "onchange './*.php', './**/*.php' -- run-s php",
+    "watch": "run-p serve watch:*",
+    "postinstall": "run-s build"
```
For many of these you can enforce local version by using, e.g. `"clean": "rimraf dist/{css/*,js/*,images/*,languages/*}",` could be `"clean": "./node_modules/.bin/rimraf dist/{css/*,js/*,images/*,languages/*}",`. If the user has a global version installed, it may cause conflicts for `npm`. All the binaries you use here should be in `./node_modules/.bin/*`.

aubreypwd commented on package.json:
```diff
@@ -2,53 +2,108 @@
+    "scss": "node-sass src/scss/style.scss -o dist/css",
+    "gutenberg": "node-sass src/scss/gutenberg-blocks-style.scss -o dist/css",
+    "lint": "eslint src/js || true",
+    "lint-scss": "stylelint src/scss/**/*.scss --syntax scss --config .stylelintrc || true",
```
We use `lint:*` style for this elsewhere, but use `lint-*` style here. In https://github.com/WebDevStudios/wd_s/pull/511 I made some of this work, but I leave the `-` style ones around in case it's in documentation, etc.

aubreypwd commented on phpcs.xml.dist:
```diff
@@ -22,8 +22,50 @@
 	<!-- Doc block alignments -->
```
This will be removed by https://github.com/WebDevStudios/wd_s/pull/511

aubreypwd reviewed (changes requested):
I have tested this in terms of linting, and got https://github.com/WebDevStudios/wd_s/pull/511 for you which we should merge in. Also there are some suggestions I have too.

aubreypwd committed:
Install and setup WDS Coding Standards (#511)

* You should be running your local version, not global.

* And, run your local version of stylelint.

* This is really lint-js and we shouldn't track vendor/

* npm run lint should do all the things.

* Remove WDS-Coding-Standards

It's outdated.

* Extend Reacty/ESNEXT

* Add some wrappers for more standard styles.

* Fix issue with priettier conflicting with WDSCS

- Update to "@webdevstudios/js-coding-standards": "~1.0.1" (stable) from beta
- Remove ESlint from root, js-coding-standards can take care of that version (was installing much newer version)
- Remove prettier, it's cleanup was suggesting things that contradicted js-coding-standards, and I don't know how to configure it properly

* Make sure our rules trump others

E.g. eslint-plugin/esnext will require padding blocks, but in ours we turn that lint off since we leave it up to developer style.

* Correct simple @return lint errors

...so all or JS is standard FTW.

* Make lint:php consistent with a wrapper.

* Install php coding standards.

* Lint all the things, except those few things.

* Lint PHP too!

* Update to 1.0.2 which allows arrowfunctions

* Also install composer stuff why not.

* composer install does this automatically.

* Use instructions from https://github.com/webdevstudios/php-coding-standards\#how-to-install

* update all config files

* dependency bump

* bump

* add prettierignore

* bump

* stylelint fixes

* use wordpress browserlist

* use wordpress standards

* remove inline ignores

* pot is in /dist

* ignore max-line-length

* bump

* update scripts to prevent infinite loop

Co-authored-by: Greg Rickaby <greg@gregrickaby.com>

coreymcollins committed:
Adds a config file for SVGO for better SVG compiling

coreymcollins committed:
Squashed commit of the following:

Merges master in

coreymcollins committed:
Fixes a stylelint issue

coreymcollins committed:
Fixes another stylelint issue with quotes

coreymcollins committed:
Sets the footer icon size
